### PR TITLE
Update CI and formating

### PR DESCRIPTION
**Pull request by petricm**
When updating the CI some formatting changes were needed to get the formatting test pass. Mostly problems when formatting was not correct.

### CODE REVIEW

**Comment by andresailer** [comment]
Is there a possibility to ensure the local clang-format version is the same as used for the check in the CI?

**Comment by petricm** [comment]
added a variable `CLANG_FORMAT_VERSION` on top level to require version of clang for formatting. If right version not found formatting targets are not created.

File: cmake/clang-cpp-checks.cmake [hunk 1]
```diff
@@ -8,6 +8,19 @@ ENDIF()
 
 # Adding clang-format check and formatter if found
 FIND_PROGRAM(CLANG_FORMAT "clang-format")
+
```
Comment (andresailer) [inline cmake/clang-cpp-checks.cmake:11]:
```suggestion
IF(NOT CLANG_FORMAT)
  RETURN()
ENDIF()
```

**Comment by andresailer** [comment]
Nice!

**Pull request merged**